### ISSUE TITLE
--buildpack flag should support a path to an archive or a url as an argument

### ISSUE THREAD

**Issue by djoyahoy** [buildpack/pack#81]
Building an app image with a remote URL or TGZ as the target of `--buildpack` does not seem to work. I'm probably doing something wrong...

See (url):
```
.root@2b65ee23-4861-433c-775d-86648d1df5d9:/tmp/build/db8923fc/buildpack# .bin/pack build test2 -p integration/fixtures/simple_app_vendored/ --buildpack https://github.com/cloudfoundry/nodejs-cnb/releases/download/v0.0.2/nodejs-cnb.tgz
2018/12/13 16:49:29 Using default builder image 'packs/samples:v3alpha2'
2018/12/13 16:49:29 Pulling builder image 'packs/samples:v3alpha2' (use --no-pull flag to skip this step)
2018/12/13 16:49:29 Pulling image 'packs/samples:v3alpha2'
2018/12/13 16:49:29 Selected run image 'packs/run:v3alpha2' from stack 'io.buildpacks.stacks.bionic'
2018/12/13 16:49:29 Pulling run image 'packs/run:v3alpha2' (use --no-pull flag to skip this step)
2018/12/13 16:49:29 Pulling image 'packs/run:v3alpha2'
*** DETECTING WITH MANUALLY-PROVIDED GROUP:
No version for 'https://github.com/cloudfoundry/nodejs-cnb/releases/download/v0.0.2/nodejs-cnb.tgz' buildpack provided, will use 'https://github.com/cloudfoundry/nodejs-cnb/releases/download/v0.0.2/nodejs-cnb.tgz@latest'
2018/12/13 16:49:31 Error: failed to detect
2018/12/13 16:49:31 Trying group of 0...
2018/12/13 16:49:31 ======== Results ========
Error: run detect container: failed with status code: 6
```
And (.tgz):
```
root@2b65ee23-4861-433c-775d-86648d1df5d9:/tmp/build/db8923fc/buildpack# .bin/pack build test2 -p integration/fixtures/simple_app_vendored/ --buildpack /tmp/nodejs-cnb.tgz
2018/12/13 16:51:39 Using default builder image 'packs/samples:v3alpha2'
2018/12/13 16:51:39 Pulling builder image 'packs/samples:v3alpha2' (use --no-pull flag to skip this step)
2018/12/13 16:51:39 Pulling image 'packs/samples:v3alpha2'
2018/12/13 16:51:39 Selected run image 'packs/run:v3alpha2' from stack 'io.buildpacks.stacks.bionic'
2018/12/13 16:51:39 Pulling run image 'packs/run:v3alpha2' (use --no-pull flag to skip this step)
2018/12/13 16:51:39 Pulling image 'packs/run:v3alpha2'
*** DETECTING WITH MANUALLY-PROVIDED GROUP:
Error: copy buildpacks to container: failed to decode buildpack.toml from "/tmp/nodejs-cnb.tgz": open /tmp/nodejs-cnb.tgz/buildpack.toml: not a directory
```

/cc @zanadar

**Comment by ekcasey** [comment]
Looking through the code and the backlog I think I misled you. `--buildpack` doesn't yet support urls or `.tgz`s. It only supports a path to a buildpack source directory or an id of a buildpack that is already on the `builder` image.

If tgz and url support is important to your workflow we can prioritize that work.

**Comment by Zanadar** [comment]
We certinaly wouldn’t complain... url would probably be most useful of those two.

**Comment by djoyahoy** [comment]
@mgibson1121 also added the help text back for this story.

**Comment by mgibson1121** [comment]
@djoyahoy I was able to successfully build using my node app using the `--buildpack` flag and targetting a .tgz.  
```ny-floater-5895:build-samples-master pivotal$ pack build someapp --buildpack ~/Downloads/nodejs-cnb-0.0.8.tgz```

However, I couldn't get the same build to work targetting the url.  Based on the title of the story, I would expect the below command to work.  Can you check my terminal output before I reject?
```
ny-floater-5895:build-samples-master pivotal$ pack build someapp --buildpack https://github.com/cloudfoundry/nodejs-cnb/releases/download/v0.0.8/nodejs-cnb.tgz
Using default builder image cloudfoundry/cnb:bionic
Pulling image index.docker.io/cloudfoundry/cnb:bionic
bionic: Pulling from cloudfoundry/cnb
Digest: sha256:175798db658a45eb6b4e707a8fde2779567650dc93544a4322036762ee463d78
Status: Image is up to date for cloudfoundry/cnb:bionic
Selected run image packs/run:0.1.0 from builder
Pulling image packs/run:0.1.0
0.1.0: Pulling from packs/run
Digest: sha256:99dee5262498d21b302321fd5d7b889c1685bfbd0ea12a64f786d91232e52278
Status: Image is up to date for packs/run:0.1.0
No version for https://github.com/cloudfoundry/nodejs-cnb/releases/download/v0.0.8/nodejs-cnb.tgz buildpack provided, will use https://github.com/cloudfoundry/nodejs-cnb/releases/download/v0.0.8/nodejs-cnb.tgz@latest
setting custom order

ERROR: failed to set custom buildpack order: no versions of buildpack https://github.com/cloudfoundry/nodejs-cnb/releases/download/v0.0.8/nodejs-cnb.tgz were found on the builder
ny-floater-5895:build-samples-master pivotal$```

**Comment by djoyahoy** [comment]
@mgibson1121  the URI for the nodejs release is: https://github.com/cloudfoundry/nodejs-cnb/releases/download/v0.0.8/nodejs-cnb-0.0.8.tgz

Your example is missing the `-0.0.8.tgz` at the end.

**Comment by mgibson1121** [comment]
@djoyahoy Still erroring out
```
ny-floater-5895:build-samples-master pivotal$ pack build someapp --buildpack https://github.com/cloudfoundry/nodejs-cnb/releases/download/v0.0.8/nodejs-cnb-0.0.8.tgz
Using default builder image cloudfoundry/cnb:bionic
Pulling image index.docker.io/cloudfoundry/cnb:bionic
bionic: Pulling from cloudfoundry/cnb
Digest: sha256:175798db658a45eb6b4e707a8fde2779567650dc93544a4322036762ee463d78
Status: Image is up to date for cloudfoundry/cnb:bionic
Selected run image packs/run:0.1.0 from builder
Pulling image packs/run:0.1.0
0.1.0: Pulling from packs/run
Digest: sha256:99dee5262498d21b302321fd5d7b889c1685bfbd0ea12a64f786d91232e52278
Status: Image is up to date for packs/run:0.1.0
No version for https://github.com/cloudfoundry/nodejs-cnb/releases/download/v0.0.8/nodejs-cnb-0.0.8.tgz buildpack provided, will use https://github.com/cloudfoundry/nodejs-cnb/releases/download/v0.0.8/nodejs-cnb-0.0.8.tgz@latest
setting custom order

ERROR: failed to set custom buildpack order: no versions of buildpack https://github.com/cloudfoundry/nodejs-cnb/releases/download/v0.0.8/nodejs-cnb-0.0.8.tgz were found on the builder
```

Using `cloudfoundry/cnb:bionic`

**Comment by djoyahoy** [comment]
@mgibson1121  cool i'll look into it

**Comment by mgibson1121** [comment]
Update:  CNB team was making changes to CI this morning, which may explain why building with the `--buidlpack` + uri build kept failing.  Will hold off on rejecting until it makes sense to test again.

**Comment by mgibson1121** [comment]
After testing with `development_pack-0.2.0-build.2-darwin`, the above issue I referenced was resolved.  Accepting this story.

**Comment by ekcasey** [comment]
fixed on `master` will ship in `v0.2.0`
